keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -143,4 +158,81 @@ void fetchingNonExistentSecretReturnsEmpty() {
     Optional<String> secret = awsSecretsManagerDefault.fetchSecret("signers-aws-integration/empty");
     assertThat(secret).isEmpty();
   }
+
+  void validateMappedSecret(
+      final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries,
+      final String secretName) {
+    final Optional<AbstractMap.SimpleEntry<String, String>> secretEntry =
+        secretEntries.stream().filter(e -> e.getKey().equals(secretName)).findAny();
+    assertThat(secretEntry).isPresent();
+    assertThat(secretEntry.get().getValue()).isEqualTo(SECRET_VALUE);
+  }
+
+  @Test
+  void multipleSecretsCanBeFetchedAndMapped() {
+    createSecret(); // 2nd
+    createSecret(); // 3rd
+
+    final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecretsList(AbstractMap.SimpleEntry::new);
+
+    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+  }
+
+  @Test
+  void multipleSecretsWithoutPrefixCanBeFetchedAndMapped() {
+    secretNamePrefix = "";
+    createSecret(); //
+    createSecret(); //
+
+    final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecretsList(AbstractMap.SimpleEntry::new);
+
+    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+    secretNamePrefix = "signers-aws-integration/";
```

Comment:
why does this need assigning at the end of the test?

Reply:
reassign `secretNamePrefix` back to its default value `"signers-aws-integration/"`